Source/EmbeddedBoundary/DistanceToEB.H

```diff
@@ -36,53 +36,90 @@ void normalize (amrex::RealVect& a) noexcept
 
-    normal[0] -= phi(i,   j  , k  ) * dxi[0] * W[1][0] * W[2][0];
-    normal[0] += phi(i+1, j  , k  ) * dxi[0] * W[1][0] * W[2][0];
-    normal[0] -= phi(i,   j+1, k  ) * dxi[0] * W[1][1] * W[2][0];
-    normal[0] += phi(i+1, j+1, k  ) * dxi[0] * W[1][1] * W[2][0];
-    normal[0] -= phi(i,   j  , k+1) * dxi[0] * W[1][0] * W[2][1];
-    normal[0] += phi(i+1, j  , k+1) * dxi[0] * W[1][0] * W[2][1];
-    normal[0] -= phi(i  , j+1, k+1) * dxi[0] * W[1][1] * W[2][1];
-    normal[0] += phi(i+1, j+1, k+1) * dxi[0] * W[1][1] * W[2][1];
-
-    normal[1] -= phi(i,   j  , k  ) * dxi[1] * W[0][0] * W[2][0];
-    normal[1] += phi(i  , j+1, k  ) * dxi[1] * W[0][0] * W[2][0];
-    normal[1] -= phi(i+1, j  , k  ) * dxi[1] * W[0][1] * W[2][0];
-    normal[1] += phi(i+1, j+1, k  ) * dxi[1] * W[0][1] * W[2][0];
-    normal[1] -= phi(i,   j  , k+1) * dxi[1] * W[0][0] * W[2][1];
-    normal[1] += phi(i  , j+1, k+1) * dxi[1] * W[0][0] * W[2][1];
-    normal[1] -= phi(i+1, j  , k+1) * dxi[1] * W[0][1] * W[2][1];
-    normal[1] += phi(i+1, j+1, k+1) * dxi[1] * W[0][1] * W[2][1];
-
-    normal[2] -= phi(i  , j  , k  ) * dxi[2] * W[0][0] * W[1][0];
-    normal[2] += phi(i  , j  , k+1) * dxi[2] * W[0][0] * W[1][0];
-    normal[2] -= phi(i+1, j  , k  ) * dxi[2] * W[0][1] * W[1][0];
-    normal[2] += phi(i+1, j  , k+1) * dxi[2] * W[0][1] * W[1][0];
-    normal[2] -= phi(i,   j+1, k  ) * dxi[2] * W[0][0] * W[1][1];
-    normal[2] += phi(i  , j+1, k+1) * dxi[2] * W[0][0] * W[1][1];
-    normal[2] -= phi(i+1, j+1, k  ) * dxi[2] * W[0][1] * W[1][1];
-    normal[2] += phi(i+1, j+1, k+1) * dxi[2] * W[0][1] * W[1][1];
+    normal[0] -= phi(ic,   j  , k  ) * dxi[0] * Wc[0][0] * W[1][0] * W[2][0];
+    normal[0] += phi(ic+1, j  , k  ) * dxi[0] * Wc[0][0] * W[1][0] * W[2][0];
+    normal[0] -= phi(ic,   j+1, k  ) * dxi[0] * Wc[0][0] * W[1][1] * W[2][0];
+    normal[0] += phi(ic+1, j+1, k  ) * dxi[0] * Wc[0][0] * W[1][1] * W[2][0];
+    normal[0] -= phi(ic,   j  , k+1) * dxi[0] * Wc[0][0] * W[1][0] * W[2][1];
+    normal[0] += phi(ic+1, j  , k+1) * dxi[0] * Wc[0][0] * W[1][0] * W[2][1];
+    normal[0] -= phi(ic,   j+1, k+1) * dxi[0] * Wc[0][0] * W[1][1] * W[2][1];
+    normal[0] += phi(ic+1, j+1, k+1) * dxi[0] * Wc[0][0] * W[1][1] * W[2][1];
+    normal[0] -= phi(ic+1, j  , k  ) * dxi[0] * Wc[0][1] * W[1][0] * W[2][0];
+    normal[0] += phi(ic+2, j  , k  ) * dxi[0] * Wc[0][1] * W[1][0] * W[2][0];
+    normal[0] -= phi(ic+1, j+1, k  ) * dxi[0] * Wc[0][1] * W[1][1] * W[2][0];
+    normal[0] += phi(ic+2, j+1, k  ) * dxi[0] * Wc[0][1] * W[1][1] * W[2][0];
+    normal[0] -= phi(ic+1, j  , k+1) * dxi[0] * Wc[0][1] * W[1][0] * W[2][1];
+    normal[0] += phi(ic+2, j  , k+1) * dxi[0] * Wc[0][1] * W[1][0] * W[2][1];
+    normal[0] -= phi(ic+1, j+1, k+1) * dxi[0] * Wc[0][1] * W[1][1] * W[2][1];
+    normal[0] += phi(ic+2, j+1, k+1) * dxi[0] * Wc[0][1] * W[1][1] * W[2][1];
+
+    normal[1] -= phi(i,   jc  , k  ) * dxi[1] * W[0][0] * Wc[1][0] * W[2][0];
```

Comment:
similar conciseness as above can be done here is well. 
note that the change will be for jc here. let me know if you need help